### PR TITLE
Am marketing modal

### PR DESCRIPTION
**Pull request by amccadams**
Adds a modal component for the marketing repo.

### CODE REVIEW

File: src/components/modal/_modal.scss [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+.modal {
+  $close-button-size: 7.2rem;
+  background-color: $color-white;
+  box-shadow: 0 .5rem 5rem rgba($color-black, .5);
+  left: 0;
+  opacity: 0;
+  position: absolute;
+  right: 0;
+  top: 0;
+//  transform: translateY(15%);
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Can we remove this?

File: src/components/modal/_modal.scss [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  @media (min-width: $min-720) {
+    left: 15%;
+    right: 15%;
+    top: 18rem;
+    width: 70%;
+  }
+
+  &--visible {
+    opacity: 1;
+//    transform: translateY(0);
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Can we remove this?

File: src/components/modal/modal_component.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    } else {
+      this.show();
+    }
+  }
+  @track("Modal Open").sendReturnValue(false)
+  show() {
+    if(this.isOpen) {
+      return Promise.all([]);
+    }
+    console.log("modal");
```
Comment (nicrocs) [inline src/components/modal/modal_component.js]: console

File: src/components/modal/_modal.scss [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+      bottom: auto;
+      left: auto;
+      max-height: calc(100vh - #{($close-button-size + $gutter * 2)});
+      padding: 4rem ($gutter * 2) ($gutter * 2);
+      position: relative;
+      right: auto;
+      top: auto;
+    }
+
+    $top-places-content-mq: (strip-units($max-width) * 10) + (strip-units($gutter) * 10 * 4);
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Can we put this after line 82? I generally like to keep all variables at the top of the block, before anything else.

File: webpack.dist.js [hunk 5]
```diff
@@ -38,7 +38,7 @@ module.exports = {
         exclude: /node_modules\/(?!rizzo|flamsteed).*/,
         query: {
             "plugins": ["transform-decorators-legacy"],
-            "presets": ["es2015", "react"]
+            "presets": ["react"]
```
Comment (nicrocs) [inline webpack.dist.js]: not sure why we got rid of this, but maybe we should try to put it back in?

File: src/components/modal/modal_component.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+   */
+  goTo(e) {
+    let el = $(e.currentTarget),
+        link = el.attr("href");
+
+    window.location = link;
+  }
+
+  toggle() {
+    if(this.isOpen){
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Nitpicking, but can we add a space between `)` and `{`?

File: src/components/modal/modal_component.js (hunk 3 above)
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Can we remove this `log`?

File: src/components/modal/modal_component.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+      this.show();
+    }
+  }
+  @track("Modal Open").sendReturnValue(false)
+  show() {
+    if(this.isOpen) {
+      return Promise.all([]);
+    }
+    console.log("modal");
+    if(this.$el.parent().length === 0){
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Space between `)` and `{`

File: src/components/modal/_modal.scss (hunk 2 above)
Comment (nicrocs) [inline src/components/modal/_modal.scss]: we can drop this altogether

File: src/components/modal/_modal.scss (hunk 1 above)
Comment (nicrocs) [inline src/components/modal/_modal.scss]: this is leftover from the top_places

File: src/components/modal/modal_component.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  onKeyup(e){
+    // ESC
+    if(e.keyCode === 27 && this.isOpen) {
+      this.hide();
+    }
+  }
+
+  handleSubmitSuccess(){
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Space between `)` and `{`

File: src/components/modal/modal_component.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    this.$form.addClass("is-hidden");
+    this.$success.removeClass("is-hidden");
+  }
+
+  submit(e){
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Space between `)` and `{`

File: src/components/modal/modal_component.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    e.preventDefault();
+    var that = this
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Missing semicolon.

@jcreamer898 How do you feel about `that = this`? Is this a spot where we can use an ES6 arrow function?

File: src/components/modal/modal_component.js [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+
+    this.$el.removeClass("modal--visible");
+
+    waitForTimeout(this.$el)
+      .then(() => {
+        this.$el.detach();
+      });
+  }
+
+  onKeyup(e){
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Space between `)` and `{`

File: src/components/modal/_modal.scss [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+    }
+  }
+
+  &__column--left {
+    margin-left: auto;
+    margin-right: auto;
+    max-width: 30rem;
+    width: 100%;
+
+
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Nitpicking, but can we remove the extra blank line here?

File: src/components/modal/_modal.scss [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  &__column--right {
+    margin-left: auto;
+    margin-right: auto;
+    max-width: 70rem;
+    width: 100%;
+
+
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Nitpicking, but can we remove the extra blank line here?

File: src/components/modal/_modal.scss [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+      }
+    }
+
+    .no-flexbox & {
+      @media (min-width: $min-960) {
+        float: left;
+        width: calc(50% - #{$gutter});
+      }
+    }
+
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Nitpicking, but can we remove the extra blank line here?

File: src/components/modal/_modal.scss [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Nitpicking, but can we remove the extra blank line here?

File: src/components/modal/_modal.scss [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+      line-height: (36.5 / 27);
+
+      @media (min-width: $min-600) {
+        font-size: 4.2rem;
+        line-height: (70 / 64);
+      }
+    }
+  }
+
+
```
Comment (thomasthesecond) [inline src/components/modal/_modal.scss]: Nitpicking, but can we remove the extra blank lines here (186 and 187)?

File: src/components/modal/ebook/ebook.hbs [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+    <option value="{{this}}">{{this}}</option>
+    {{/each}}
+    </select>
+    <input type="submit" value="Submit">
+    </form>
+  {{/if}}
+  {{ebook.terms}}
+<div class="js-success is-hidden">
+Thank you!
+</div>
```
Comment (thomasthesecond) [inline src/components/modal/ebook/ebook.hbs]: Can we indent 30-32 to be on the same column as 29?

File: src/components/modal/ebook/ebook.hbs [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+</div>
+
+
+
+
```
Comment (thomasthesecond) [inline src/components/modal/ebook/ebook.hbs]: Remove extra blank lines, 35-37. Our editorconfig will require one blank line at the end of files.

File: src/components/modal/_modal.scss (hunk 4 above)
Comment (nicrocs) [inline src/components/modal/_modal.scss]: I'm not sure what this is supposed to do, but looks like leftover from top_places. Looks like it has something to with adjusting spacing with media queries

**Comment by nicrocs** [comment]
just a few things to clean up. otherwise looks good

File: src/components/modal/modal_component.js [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  handleSubmitSuccess() {
+    this.$form.addClass("is-hidden");
+    this.$success.removeClass("is-hidden");
+  }
+
+  submit(e) {
+    e.preventDefault();
+    var that = this;
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]: Let's remove `var that = this;` and use ES6 arrow functions instead. ES6 arrow functions bind `this` so we don't need to set `this` to a variable. See code below.

File: src/components/modal/modal_component.js [hunk 20]
```diff
@@ -0,0 +1,114 @@
...
+    $.post(this.$form.attr("action"), this.$form.serialize())
+      .done(function() {
+        that.handleSubmitSuccess();
+      })
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]:
```
.done(() => {
  this.handleSubmitSuccess();
}
```

File: src/components/modal/modal_component.js [hunk 21]
```diff
@@ -0,0 +1,114 @@
...
+      .fail(function(xhr) {
+        if (xhr.status === 409) {
+          that.handleSubmitSuccess();
```
Comment (thomasthesecond) [inline src/components/modal/modal_component.js]:
```
.fail((xhr) => {
  if (xhr.status === 409) {
    this.handleSubmitSuccess();
```

**Pull request merged**